controller_manager/src/controller_manager.cpp

```diff
@@ -1226,15 +1226,26 @@ void ControllerManager::list_controllers_srv_cb(
   // lock controllers
   std::lock_guard<std::recursive_mutex> guard(rt_controllers_wrapper_.controllers_lock_);
   const std::vector<ControllerSpec> & controllers = rt_controllers_wrapper_.get_updated_list(guard);
-  response->controller.resize(controllers.size());
+  response->controller.reserve(controllers.size());
```

Comment:
since `response->controller` is not used in the next loop anymore I think it'd be less error-prone to move the `reserve()` closer to where `response->controller.push_back()` occurs

Reply:
I moved it to the beginning of the loop that pushes to  it.